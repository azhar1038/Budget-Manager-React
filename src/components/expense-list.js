import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './expense-list-item'
import selectExpenses from '../selectors/expenses'

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses Found</p>
            ):(
                props.expenses.map((expense)=>(
                    <ExpenseListItem
                        key={expense.id}
                        {...expense}
                    />
                ))
            )
        }
    </div>
)

const mapStateToProps = (state)=>({
    expenses: selectExpenses(state.expenses, state.filters)
})

export default connect(mapStateToProps)(ExpenseList)