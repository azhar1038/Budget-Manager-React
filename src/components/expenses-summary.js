import React from 'react'
import { connect } from 'react-redux'
import selectExpenses from '../selectors/expenses'
import totalExpenses from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpensesSummary = ({expenseCount, expensesTotal})=>{
    const expenseWord = expenseCount > 1? 'expenses': 'expense'
    const formattedTotal = numeral(expensesTotal/100).format('0,0.00')
    return(
        <div>
            Viewing {expenseCount} {expenseWord} totalling â‚¹{formattedTotal}
        </div>
    )
}

const mapStateToProps = (state)=>{
    const expenses = selectExpenses(state.expenses, state.filters)
    return {
        expenseCount: expenses.length,
        expensesTotal: totalExpenses(expenses)
    }
}

export default connect(mapStateToProps)(ExpensesSummary)