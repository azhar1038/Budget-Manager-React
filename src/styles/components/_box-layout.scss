.box-layout{
    align-items: center;
    background: url('/images/bg.jpg');
    background-size: cover;
    display: flex;
    height: 100vh;
    justify-content: center;
    width: 100vw;
}

.box-layout__box{
    background: fade-out($color: white, $amount: .25);
    border-radius: 4px;
    padding: $l-size $m-size;
    text-align: center;
    max-width: 25rem;
    margin: $s-size;
}

.box-layout__title{
    margin: 0 0 $m-size 0;
    line-height: 1;
}