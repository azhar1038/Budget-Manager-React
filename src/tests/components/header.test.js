import React from 'react'
import { shallow } from 'enzyme'
import { Header } from '../../components/header'

let startLogout, wrapper

beforeEach(()=>{
    startLogout = jest.fn()
    wrapper = shallow(<Header startLogout={startLogout} />)
})

test('Should render header correctly',()=>{
    expect(wrapper).toMatchSnapshot()
})

test('Should call startLogout on button click', ()=>{
    wrapper.find('button').simulate('click')
    expect(startLogout).toHaveBeenCalled()
})