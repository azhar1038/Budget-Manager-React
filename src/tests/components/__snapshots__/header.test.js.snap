// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header correctly 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Add Expense
  </NavLink>
  <button
    onClick={[MockFunction]}
  >
    Logout
  </button>
</div>
`;
